### ISSUE TITLE
Transfer inserting extra carriage returns on windows

### ISSUE THREAD

**Issue by mr-martian** [apertium/apertium#172]
Apparently inputs containing `\r\n` on Windows are coming out of `apertium-transfer` as `\r\r\n`. I was not able to reproduce this on Linux.

**Comment by TinoDidriksen** [comment]
Text-mode output translates `\n` to `\r\n`. But input should also translate `\r\n` to `\n`, thus preventing such double outputs.

It is fully expected that outputting `\r\n` will yield `\r\r\n`, but internally there shouldn't be any `\r` in the data.

**Comment by rmlockwood** [comment]
I'm on Windows, so naturally there's going to be \r's in the data when I create an input file that I want apertium-transfer to process. Are you saying I need to remove \r's before using apertium-transfer?

**Comment by TinoDidriksen** [comment]
No, this should be handled automatically. The fact that it's not means we have a weird bug.

**Comment by rmlockwood** [comment]
Any progress on this?

**Comment by mr-martian** [comment]
`InputFile` always opens files in binary mode, could that be preventing the replacement? Maybe `InputFile` should just skip `\r` on Windows? Or everywhere before `\n`?

**Comment by mr-martian** [comment]
Oh, but `apertium-transfer` is one of the few places that opens output files in text mode. I'll change that and see if it fixes this.
